src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/cache_tuning/validator/FieldDataCacheDeciderValidator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.tests.cache_tuning.validator;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ModifyCacheMaxSizeAction;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.framework.api.IValidator;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.actions.PersistedAction;
+import org.junit.Assert;
+
+public class FieldDataCacheDeciderValidator implements IValidator {
+    long startTime;
+
+    public FieldDataCacheDeciderValidator() {
+        startTime = System.currentTimeMillis();
+    }
+
+    /**
+     * {"actionName":"ModifyCacheMaxSize",
+     * "resourceValue":10,
+     * "timestamp":"1599257910923",
+     * "nodeId":{node1},
+     * "nodeIp":{1.1.1.1},
+     * "actionable":1,
+     * "coolOffPeriod": 300000
+     * "muted": 1
+     * "summary": Update [FIELD_DATA_CACHE] capacity from [10000] to [100000] on node [DATA_0]
+     */
+    @Override
+    public boolean checkDbObj(Object object) {
+        if (object == null) {
+            return false;
+        }
+        PersistedAction persistedAction = (PersistedAction) object;
+        return checkPersistedAction(persistedAction);
+    }
+
+    /**
+     * {"actionName":"ModifyCacheMaxSize",
+     * "resourceValue":10,
+     * "timestamp":"1599257910923",
+     * "nodeIds":{node1},
+     * "nodeIps":{1.1.1.1},
+     * "actionable":1,
+     * "coolOffPeriod": 300000
+     * "muted": 1
+     * "summary": Update [FIELD_DATA_CACHE] capacity from [10000] to [100000] on node [DATA_0]
+     *
+     */
+    private boolean checkPersistedAction(final PersistedAction persistedAction) {
```

Comment:
Instead of this, we should write the `ModifyCacheMaxSizeAction.from_summar()` method, and validate that action here.

Reply:
Added from_summary method, and using that method for validations.